src/python/pants/core/util_rules/partitions.py

```diff
@@ -182,3 +202,78 @@ async def partitioner(
         )
 
     return collect_rules(locals())
+
+
+@memoized
+def _partition_per_input_field_sets_partitioner_rules(cls) -> Iterable:
+    """Returns a rule that implements a "partitioner" for `PartitionFieldSetsRequest`, which returns
+    a single-element partition per input."""
+
+    @rule(
+        _param_type_overrides={
+            "request": cls.PartitionRequest,
+            "subsystem": cls.tool_subsystem,
+        }
+    )
+    async def partitioner(
+        request: _PartitionFieldSetsRequestBase, subsystem: SkippableSubsystem
+    ) -> Partitions:
```

Comment:
This should be typed, `Partitions[str, FieldSet]`. However `str` is not a valid `Partitions` key, which is relevant :smile: 

You might need to invent a tiny class that wraps an address, or provide a `description` property on `Address`.

Reply:
See #17259 for my counter-suggestion 😉